[DOCS] Relocate thread pools content (#55814)
elastic/elasticsearch: docs/reference/modules/threadpool.asciidoc

[[modules-threadpool]]
=== Thread pools

> 
> A node holds several thread pools in order to improve how threads memory consumption
> are managed within a node. Many of these pools also have queues associated with them,
> which allow pending requests to be held instead
> of discarded.

Reviewer: ```suggestion

A node uses several thread pools to manage memory consumption. 
Queues associated with many of the thread pools enable pending requests 
to be held instead of discarded. 
```
